### PR TITLE
Add boolean getters to http headers and query params

### PR DESCRIPTION
**Pull request by minwoox**
Motivation:
It will be nice if we provide a way to get a boolean value easily from `QueryParams` and `HttpHeaders`.
Currently, we have getters for int, long, double, and float. We don't have to provide for all primitive types,
but boolean parameters are used pretty commonly.

Modifications:
- Add boolean getters to `QueryParams` and `HttpHeaders`.

Result:
- You can easily get a boolean value from `QueryParams` and `HttpHeaders`.

### CODE REVIEW

**Review by ikhoon — APPROVED** [review]
Thanks! @minwoox

File: core/src/main/java/com/linecorp/armeria/common/StringMultimap.java [hunk 1]
```diff
@@ -957,6 +988,20 @@ private String remove0(int h, int i, IN_NAME name) {
...
+    private static Boolean toBoolean(@Nullable String v) {
+        if (v == null) {
+            return null;
+        }
+        if ("true".equalsIgnoreCase(v)) {
+            return true;
+        }
+        if ("false".equalsIgnoreCase(v)) {
+            return false;
+        }
```
Comment (trustin) [inline core/src/main/java/com/linecorp/armeria/common/StringMultimap.java:1001]: Should we convert a number to a boolean, like 0 to false and other to true? What do we do for other values like on/off and yes/no? What's the widely accepted sane behavior among other frameworks?

Comment (trustin) [reply core/src/main/java/com/linecorp/armeria/common/StringMultimap.java:1001]: Shouldn't we handle `0` and `1` for consistency with annotated services?

Comment (minwoox) [reply core/src/main/java/com/linecorp/armeria/common/StringMultimap.java:1001]: Yeah, I will do. Just waiting for the result of https://github.com/line/armeria/issues/3619 and I think I can do it now. 😄

Comment (minwoox) [reply core/src/main/java/com/linecorp/armeria/common/StringMultimap.java]: Fixed. 😉

File: core/src/main/java/com/linecorp/armeria/common/StringMultimap.java [hunk 2]
```diff
@@ -431,6 +458,11 @@ public final boolean containsObject(IN_NAME name, Object value) {
         return contains(name, fromObject(value));
     }
 
+    @Override
+    public final boolean containsBoolean(IN_NAME name, boolean value) {
+        return contains(name, String.valueOf(value));
```
Comment (trustin) [inline core/src/main/java/com/linecorp/armeria/common/StringMultimap.java:463]: This seems incorrect? Should match `1` and `0` as well as `true` and `false`, right?

File: core/src/test/java/com/linecorp/armeria/common/HttpHeadersBaseTest.java [hunk 3]
```diff
@@ -93,6 +93,24 @@ void testGetLastOperationsWithMultipleValues() {
         assertThat(headers.getLast("Foo")).isEqualTo("2");
     }
 
+    @Test
+    void testGetBooleanOperation() {
+        final HttpHeadersBase headers = newEmptyHeaders();
+        headers.add("foo", "1");
+        assertThat(headers.getBoolean("foo")).isNull();
+        assertThat(headers.getBoolean("foo", true)).isTrue();
```
Comment (trustin) [inline core/src/test/java/com/linecorp/armeria/common/HttpHeadersBaseTest.java:101]: This needs an update, right?

File: core/src/test/java/com/linecorp/armeria/common/QueryParamsBaseTest.java [hunk 4]
```diff
@@ -89,6 +89,24 @@ void testGetLastOperationsWithMultipleValues() {
         assertThat(params.getLast("Foo")).isEqualTo("2");
     }
 
+    @Test
+    void testGetBooleanOperation() {
+        final QueryParamsBase params = newEmptyParams();
+        params.add("foo", "1");
+        assertThat(params.getBoolean("foo")).isNull();
+        assertThat(params.getBoolean("foo", true)).isTrue();
```
Comment (trustin) [inline core/src/test/java/com/linecorp/armeria/common/QueryParamsBaseTest.java:97]: and this one as well.

File: core/src/main/java/com/linecorp/armeria/common/StringMultimap.java (hunk 2 above)
Comment (minwoox) [reply core/src/main/java/com/linecorp/armeria/common/StringMultimap.java:463]: But then we have to add a method like
`containsBoolean(IN_NAME name, String value) {...}`
Do we need that? 🤔

Comment (minwoox) [reply core/src/main/java/com/linecorp/armeria/common/StringMultimap.java:463]: Let me make it case insensitive by the way. 😄

Comment (trustin) [reply core/src/main/java/com/linecorp/armeria/common/StringMultimap.java:463]: No. You can do `getBoolean(name)` and then compare.

Comment (minwoox) [reply core/src/main/java/com/linecorp/armeria/common/StringMultimap.java:463]: Thanks! Now I get it. 😓

File: core/src/test/java/com/linecorp/armeria/common/HttpHeadersBaseTest.java (hunk 3 above)
Comment (minwoox) [reply core/src/test/java/com/linecorp/armeria/common/HttpHeadersBaseTest.java]: updated. PTAL 🙇

File: it/jackson-provider/src/test/java/com/linecorp/armeria/common/CustomJacksonModuleProvider.java [hunk 5]
```diff
@@ -22,8 +22,6 @@
 import com.fasterxml.jackson.module.kotlin.KotlinModule;
 import com.google.common.collect.ImmutableList;
 
-import com.linecorp.armeria.common.JacksonModuleProvider;
-
```
Comment (ikhoon) [inline it/jackson-provider/src/test/java/com/linecorp/armeria/common/CustomJacksonModuleProvider.java:26]: 😱

**Review by trustin — APPROVED** [review]

File: core/src/main/java/com/linecorp/armeria/common/StringMultimap.java [hunk 6]
```diff
@@ -431,6 +464,12 @@ public final boolean containsObject(IN_NAME name, Object value) {
         return contains(name, fromObject(value));
     }
 
+    @Override
+    public final boolean containsBoolean(IN_NAME name, boolean value) {
+        final Boolean v = getBoolean(name);
+        return v != null && v == value;
+    }
+
```
Comment (injae-kim) [inline core/src/main/java/com/linecorp/armeria/common/StringMultimap.java:472]:
```
    @Override
    public final boolean containsInt(IN_NAME name, int value) {
        return contains(name, String.valueOf(value));
    }
```
`containsBoolean` should use `contains()` same with other `contains*`? 🤔
Fix PR: https://github.com/line/armeria/pull/5319

File: core/src/main/java/com/linecorp/armeria/common/StringMultimap.java (hunk 2 above)
Comment (injae-kim) [reply core/src/main/java/com/linecorp/armeria/common/StringMultimap.java]: > No. You can do getBoolean(name) and then compare.

Hmm If we use `getBoolean` inside of `containsBoolean`, it checks only first entry using `get(name)`?
Maybe `containsBoolean` should check all entries like other `contains*` 🤔

Please check my Fix PR: https://github.com/line/armeria/pull/5319 for details 🙇

**Pull request merged**